docs/source/contents/postprocessing.rst

```diff
@@ -1,48 +1,108 @@
 Pairwise Processing
 =============================
 
+Note, this part of the library is under construction.
+
+Interfaces
+================================================================================================
 .. toctree::
    :titlesonly:
 
 .. contents::
    :local:
 
-Note, this part of the library is under construction.
-
-
-IPostprocessor
+IDistancesPostprocessor
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-.. autoclass:: oml.interfaces.postprocessor.IPostprocessor
+.. autoclass:: oml.interfaces.retrieval.IDistancesPostprocessor
     :undoc-members:
     :show-inheritance:
 
     .. automethod:: process
 
-
-IPairwiseDistanceModel
+IPairwiseModel
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-.. autoclass:: oml.interfaces.models.IPairwiseDistanceModel
+.. autoclass:: oml.interfaces.models.IPairwiseModel
     :undoc-members:
     :show-inheritance:
 
     .. automethod:: forward
 
+IPairsDataset
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+.. autoclass:: oml.interfaces.datasets.IPairsDataset
+    :undoc-members:
+    :show-inheritance:
+
+    .. automethod:: __init__
+    .. automethod:: __getitem__
+
+Implementations
```

Comment:
readthedocs looks ugly, it should be nested

Reply:
reworked